### PR TITLE
[BugFix] Fix array_difference<largint> error

### PR DESCRIPTION
**Pull request by Seaven**
Why I'm doing:

What I'm doing:
the register info:
```
    [150160, 'array_difference',  'ARRAY_BIGINT',       ['ARRAY_BOOLEAN'],      'ArrayFunctions::array_difference<TYPE_BOOLEAN>'],
    [150161, 'array_difference',  'ARRAY_BIGINT',       ['ARRAY_TINYINT'],      'ArrayFunctions::array_difference<TYPE_TINYINT>'],
    [150162, 'array_difference',  'ARRAY_BIGINT',       ['ARRAY_SMALLINT'],     'ArrayFunctions::array_difference<TYPE_SMALLINT>'],
    [150163, 'array_difference',  'ARRAY_BIGINT',       ['ARRAY_INT'],          'ArrayFunctions::array_difference<TYPE_INT>'],
    [150164, 'array_difference',  'ARRAY_BIGINT',       ['ARRAY_BIGINT'],       'ArrayFunctions::array_difference<TYPE_BIGINT>'],
    [150165, 'array_difference',  'ARRAY_LARGEINT',     ['ARRAY_LARGEINT'],     'ArrayFunctions::array_difference<TYPE_LARGEINT>'],
    [150166, 'array_difference',  'ARRAY_DOUBLE',       ['ARRAY_FLOAT'],        'ArrayFunctions::array_difference<TYPE_FLOAT>'],
    [150168, 'array_difference',  'ARRAY_DECIMALV2',    ['ARRAY_DECIMALV2'],    'ArrayFunctions::array_difference<TYPE_DECIMALV2>'],
    [1501681, 'array_difference', 'ARRAY_DECIMAL32',    ['ARRAY_DECIMAL32'],    'ArrayFunctions::array_difference<TYPE_DECIMAL32>'],
    [1501682, 'array_difference', 'ARRAY_DECIMAL64',    ['ARRAY_DECIMAL64'],    'ArrayFunctions::array_difference<TYPE_DECIMAL64>'],
    [1501683, 'array_difference', 'ARRAY_DECIMAL128',   ['ARRAY_DECIMAL128'],   'ArrayFunctions::array_difference<TYPE_DECIMAL128>'],
    [150167, 'array_difference',  'ARRAY_DOUBLE',       ['ARRAY_DOUBLE'],       'ArrayFunctions::array_difference<TYPE_DOUBLE>'],
```

Largeint should return Largeint, not bigint

Fixes #issue
https://github.com/StarRocks/StarRocksTest/issues/6082
## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java [hunk 1]
```diff
@@ -221,7 +221,8 @@ private static void createOutputFragment(PlanFragment inputFragment, ExecPlan ex
 
     private static boolean enableComputeNode(ExecPlan execPlan) {
         boolean preferComputeNode = execPlan.getConnectContext().getSessionVariable().isPreferComputeNode();
-        if (preferComputeNode) {
+        int computeNode = execPlan.getConnectContext().getSessionVariable().getUseComputeNodes();
+        if (preferComputeNode || computeNode <= 0) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java:225]: is this condtion right?

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java:225]: right, we should more strict

**Review by kangkaisen — APPROVED** [review]

**Review by fzhedu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by fzhedu** [comment]
https://github.com/Mergifyio backport branch-2.5

**Pull request merged**